src/crush/CrushLocation.cc

```diff
@@ -93,9 +91,8 @@ int CrushLocation::update_from_hook()
 
   std::string out;
   bl.begin().copy(bl.length(), out);
-  out.erase(out.find_last_not_of(" \n\r\t")+1);
+  boost::algorithm::trim_if(out, boost::algorithm::is_any_of(" \n\r\t"));
```

Comment:
I don't think these have the same behavior.  What is the behavior you expect?

Reply:
yeah, a little different, `trim_if` will trim both the left and right side of the string.
e.g.
```
string out= "\n\r0\n\r123\n\r"
out.erase(out.find_last_not_of(" \n\r\t")+1); //out is "\n\r0\n\r123"
boost::algorithm::trim_if(out, boost::algorithm::is_any_of(" \n\r\t")); // out is "0\n\r123"
```
Does it need to change `boost::algorithm::trim_if` to `boost::algorithm::trim_right_if`?

Reply:
Oh, I actually misunderstood find_last_not_of.  Yeah, boost::algorithm::trim_right_if should be perfect.